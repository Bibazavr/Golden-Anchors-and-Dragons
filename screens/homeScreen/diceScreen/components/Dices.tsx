import * as React from "react";
import {StyleProp, StyleSheet, View, ViewStyle} from "react-native";
import {Button, Text} from "react-native-elements";

import {diceDrop} from "../utils/dice";

export const Dice = (props: { facets: number, style?: StyleProp<ViewStyle> }) => {
    const [value, setValue] = React.useState<number>(props.facets);

    const drop = () => setValue(diceDrop(props.facets));

    return <View style={stylesDices.containerDice}>
        <Button title={value.toString()} buttonStyle={[stylesDices.dice, props?.style]} onPress={drop}
                testID={'diceButton'}/>
        <Text>d{props.facets}</Text>
    </View>
}

export const Dices = () => {
    return <View style={stylesDices.container}>
        <Dice facets={4} style={stylesDices.dice4}/>
        <Dice facets={6} style={stylesDices.dice6}/>
        <Dice facets={8} style={stylesDices.dice8}/>
        <Dice facets={10} style={stylesDices.dice10}/>
        <Dice facets={12} style={stylesDices.dice12}/>
        <Dice facets={20} style={stylesDices.dice20}/>
        <Dice facets={100} style={stylesDices.dice100}/>
    </View>
};

const stylesDices = StyleSheet.create({
    container: {
        flexDirection: "row",
        flexWrap: "wrap",
        alignItems: "center",
        justifyContent: "center",
    },
    containerDice: {
        margin: 10,
        alignItems: "center",
        justifyContent: "center"
    },
    dice: {
        height: 50,
        width: 50,
    },
    dice4: {
        backgroundColor: '#7851a9',
    },
    dice6: {
        backgroundColor: '#7851a9',
    },
    dice8: {
        backgroundColor: '#7851a9',
    },
    dice10: {
        backgroundColor: '#7851a9',
    },
    dice12: {
        backgroundColor: '#7851a9',
    },
    dice20: {
        backgroundColor: '#7851a9',
    },
    dice100: {
        backgroundColor: '#7851a9',
    },
});
