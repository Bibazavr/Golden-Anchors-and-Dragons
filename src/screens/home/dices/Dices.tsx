import * as React from "react";
import {StyleSheet, View} from "react-native";

import {Dices} from "./components/Dices";


export const DiceScreen = () => {
    return (
        <View style={stylesDiceScreen.container}>
            <Dices/>
        </View>
    )
}

const stylesDiceScreen = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "center",
        backgroundColor: '#fff',
    },
});
