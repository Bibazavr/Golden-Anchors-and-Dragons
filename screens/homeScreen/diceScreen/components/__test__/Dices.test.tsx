import * as React from "react";
import {fireEvent, render} from "@testing-library/react-native";

import {Dice, Dices} from "../Dices";

describe("dices", () => {
    it(`drop4`, () => {
        const page = render(<Dice facets={4}/>);

        const dice = page.getByTestId("diceButton");

        fireEvent.press(dice);

        expect(page.asJSON()).not.toBeUndefined();
    });
    it('renders correctly', () => {
        const page = render(<Dices/>);

        expect(page.asJSON()).toMatchSnapshot()
    })
})
