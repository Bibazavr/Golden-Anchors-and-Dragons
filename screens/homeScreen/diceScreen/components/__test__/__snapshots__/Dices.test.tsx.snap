// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dices renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="diceButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#7851a9",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              4
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={Object {}}
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        d
        4
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="diceButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#7851a9",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              6
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={Object {}}
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        d
        6
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="diceButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#7851a9",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              8
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={Object {}}
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        d
        8
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="diceButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#7851a9",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              10
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={Object {}}
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        d
        10
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="diceButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#7851a9",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              12
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={Object {}}
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        d
        12
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="diceButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#7851a9",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              20
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={Object {}}
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        d
        20
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="diceButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#7851a9",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
                "width": 50,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              100
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={Object {}}
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        d
        100
      </Text>
    </View>
  </View>
</View>
`;
